src/nvim/api/buffer.c

```diff
@@ -291,7 +291,11 @@ void buffer_set_line_slice(Buffer buffer,
   // Only adjust marks if we managed to switch to a window that holds
   // the buffer, otherwise line numbers will be invalid.
   if (save_curbuf == NULL) {
-    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
+    /// WCONV: extra is ssize_t, representing lines added or removed, so
+    /// WCONF: logically it should be < MAXLNUM, which is INT32_MAX
+    /// WCONF: Add an assertion to be sure
+    assert( -MAXLNUM <= extra && extra <= MAXLNUM);
```

Comment:
Is this assert really called for here? I mean if we managed to insert more than MAXLNUM lines, we would already have wreaked havoc _before_ reaching this line. For whatever reason we are checking `lnum > LONG_MAX` before `ml_append`, if we check `lnum >= MAXLNUM` instead we should be fine. (Or even better: check if `buf->b_ml.ml_line_count+new_len-old_len < MAXLNUM` before even starting)